unblob/handlers/executable/elf.py

```diff
@@ -190,13 +234,39 @@ def get_end_offset(
             section_headers_end, program_headers_end, last_section_end, last_program_end
         )
 
+    def is_signed_kernel(self, file: File, header, offset: int) -> bool:
+        return (
+            header.e_type == lief.ELF.E_TYPE.RELOCATABLE.value
+            and file[offset : offset + 1] == ASN1_SEQUENCE_START  # noqa: E203
+        )
+
     def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]:
         endian = self.get_endianness(file, start_offset)
         file.seek(start_offset, io.SEEK_SET)
         header = self.parse_header(file, endian)
         if not self.is_valid_header(header):
             return
         end_offset = self.get_end_offset(file, start_offset, header, endian)
+
+        if self.is_signed_kernel(file, header, end_offset):
+            length_bytes = file[end_offset : end_offset + ASN1_PREFIX_LEN]  # noqa: E203
+            signature_length = get_asn1_sequence_length(length_bytes)
+            if signature_length:
```

Comment:
do we want to do any signature sanity check? It is not very likely, but if the kernel module has `0x30` byte after it's end we most likely would pick it up as a signature.

We would be able to check the signature.sig_len value which should contain the signature len.

Maybe a better alternative would be to search for the `KERNEL_MODULE_SIGNATURE_FOOTER` after the end of the ELF file and check the signature info to see if the size would fit. This way we could avoid the asn1 length parsing as well if we want

Reply:
This was my initial idea and it makes even more sense knowing about `module_signature`.

We could:

1. search for `KERNEL_MODULE_SIGNATURE_FOOTER`
2. if found, seek to the start of `module_signature`
3. parse `module_signature` and check that `signature.sig_len` corresponds to the end offset of our signature footer minus the end offset of current ELF file.

This way we can drop the ASN1 sequence parsing.

Reply:
Just pushed my changes, let me know what you think.

Reply:
Added a few test cases for non-happy paths and simplified the code to use `iterate_patterns`